lib/Connection.js

```diff
@@ -214,6 +216,16 @@ class CDPSession extends EventEmitter {
         callback.reject(rewriteError(callback.error, `Protocol error (${callback.method}): ${object.error.message} ${object.error.data}`));
       else
         callback.resolve(object.result);
+    } else if (object.method === 'Target.receivedMessageFromTarget') {
+      const session = this._sessions.get(object.params.sessionId);
+      if (session)
+        session._onMessage(object.params.message);
+    } else if (object.method === 'Target.detachedFromTarget') {
+      const session = this._sessions.get(object.params.sessionId);
+      if (session)
+        session._onClosed();
+      this._sessions.delete(object.params.sessionId);
+      this.emit(object.method, object.params);
```

Comment:
why this emit?

Reply:
We still want to send the event to people interacting with the raw protocol. But I see that we need it above in recievedMessageFromTarget as well.